src/index.ts

```diff
@@ -60,44 +60,48 @@ function vitePreprocessor(userConfigPath?: string): CypressPreprocessor {
       },
     }
 
-    const watcher = await build({
-      configFile: userConfigPath,
-      ...defaultConfig,
-    })
-
-    if (shouldWatch && isWatcher(watcher)) {
-      watcher.on('event', (event) => {
-        debug('Watcher %s for file %s', event.code, filePath)
-
-        if (event.code === 'END') {
-          file.emit('rerun')
-        }
-
-        if (event.code === 'ERROR') {
-          console.error(event)
-        }
-      })
-
-      file.on('close', () => {
-        delete cache[filePath]
-        watcher.close()
-
-        debug('File %s closed.', filePath)
-      })
-    }
-
     cache[filePath] = outputPath
     debug('Bundle for file %s cached at %s', filePath, outputPath)
 
-    return outputPath
+    const watcher = (await build({
+      configFile: userConfigPath,
+      ...defaultConfig,
+    })) as BuildResult
```

Comment:
I had to add `as BuildResult` to get it to build so I could test, but I have no idea why because the type returned from build matches `BuildResult`.

Reply:
I think this is a because of a mismatch in Rollup versions.